src/main/kotlin/com/colivery/serviceaping/rest/v1/services/AdminRestService.kt

```diff
@@ -0,0 +1,39 @@
+package com.colivery.serviceaping.rest.v1.services
+
+import com.colivery.serviceaping.mapping.toUserResource
+import com.colivery.serviceaping.persistence.entity.UserEntity
+import com.colivery.serviceaping.persistence.repository.UserRepository
+import com.colivery.serviceaping.rest.v1.dto.user.PatchUserAdminDto
+import com.colivery.serviceaping.rest.v1.resources.UserResource
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.server.reactive.ServerHttpResponse
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+import reactor.core.publisher.Mono
+import javax.transaction.Transactional
+import javax.validation.Valid
+
+@RestController
+@PreAuthorize("hasRole('ROLE_ADMIN')")
+@Transactional
+@RequestMapping("/v1/admin", produces = [MediaType.APPLICATION_JSON_VALUE])
+class AdminRestService(private val userRepository: UserRepository) {
+
+    @PatchMapping("/users/{userId}")
+    fun updateUserAdmin(@PathVariable userId: String, @RequestBody @Valid userPatch: PatchUserAdminDto,
+                        response: ServerHttpResponse)
+            : Mono<UserResource> {
+
+        var user = userRepository.findByFirebaseUid(userId)
+        if (user == null) {
+            response.statusCode = HttpStatus.NOT_FOUND
+            return Mono.empty()
+        }
+
+        user.role = UserEntity.Role.valueOf(userPatch.role!!)
```

Comment:
Why do we need the `!!` operator here? Can't we make the role non-null in the DTO instead?

Reply:
I think the problem here is my experience with Kotlin: the issue is that the javax validations happen only after the object is created, so if we set the role field to be "kotlin not nullable", we will get some jackson exceptions, because jackson won't be able to create the object.. and this all would happen before the new validation component even gets triggered, so we would need to treat for it in a different point in the code.. in the end I though it would be better to treat all the validations in the same point in the code, so make the field "kotlin nullable", let jackson create the object and let the validator validate it.. but again, there might be better ways to do it in Kotlin